docker-compose.vm.yml

```diff
@@ -0,0 +1,10 @@
+version: '3.7'
+services:
+  node1:
+    image: waggle/node-platforms:latest
+    build: 
+      context: node-platforms/vm
+      dockerfile: Dockerfile
+      args:
```

Comment:
Yes, just use the .env var at build time.

This is seperate from having multiple nodes. This was more about using this pattern for integration steps. For example when we want to test with beehive.

I did not want to go that route since I am only going to test at most two locally. If I want to go more than 2, I will start testing in kubernetes.

The way I demoed it when you were gone was to use the scale option in docker-compose. This will be done after I adjust the values in the .env file (VSN and  NODE_ID).

```
docker-compose -f docker-compose.yml -f docker-compose.vm.yml up --build --scale node1=2 -d
```

This will generate `node1_1` and `node1_2`

It does overwrite the values from `node1_1` but since it already registered with beekeeper the db has the previous node id before I modified it. 

[SAGE-1382](https://jira.cels.anl.gov/browse/SAGE-1382) aims to figure out the ssh keys so that it works for multiple nodes so that's why I did not add the scale stuff to this PR. I might have to add something similar to what you are proposing in `docker-compose.nodes.yaml` if the nodeid is being checked.

Reply:
Alright the PR is ready. Fix all the problems and now works. Added your suggestions.

Reply:
Need to add another workflow to run the unit-test for the docker-compose overlay.